### PR TITLE
App Configuration - Data Loader

### PR DESCRIPTION
**Pull request by mrm9084**
# Description

Updates App Configuration to use the Data Loader.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationBootstrapRegistrar.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import com.azure.spring.cloud.appconfiguration.config.implementation.properties.AppConfigurationProviderProperties;
+
+class AppConfigurationBootstrapRegistrar {
+
+    static void register(ConfigDataLocationResolverContext context, Binder binder,
+        AppConfigurationProperties properties, AppConfigurationProviderProperties appProperties) {
+        AppConfigurationKeyVaultClientFactory keyVaultClientFactory = appConfigurationKeyVaultClientFactory(
+            appProperties, context,
+            binder);
+        AppConfigurationReplicaClientsBuilder replicaClientsBuilder = replicaClientBuilder(context, binder,
```
Comment (mrm9084) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationBootstrapRegistrar.java:22]: Move inside registerIfAbsent so it is only created once.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/pom.xml [hunk 2]
```diff
@@ -142,12 +137,12 @@
           <rules>
             <bannedDependencies>
               <includes>
+				<include>jakarta.annotation:jakarta.annotation-api:[2.1.1]</include> <!-- {x-include-update;jakarta.annotation:jakarta.annotation-api;external_dependency} -->
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/pom.xml:140]: Seems like the only annotations we are using is `PostConstruct`. I'd recommned that we use `implements InitializingBean` instead, so we don't need to depend on the `jakarta.annotations` directly.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigDataLoader.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    private StateHolder storeState = new StateHolder();
+    
+    private List<AppConfigurationPropertySource> sources = new ArrayList<>();
+
+    public AppConfigDataLoader(DeferredLogFactory logFactory) {
+        this.logger = logFactory.getLog(getClass());
+    }
+
+    @Override
+    public ConfigData load(ConfigDataLoaderContext context, AppConfigDataResource resource)
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigDataLoader.java:47]: No `IOException` is thrown.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigDataLoader.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                if (propertySource instanceof AppConfigurationFeatureManagementPropertySource) {
+                    watchKeysFeatures.addAll(
+                        ((AppConfigurationFeatureManagementPropertySource) propertySource).getFeatureFlagSettings());
+                }
+            }
+        }
+        return watchKeysFeatures;
+    }
+
+    private StateHolder failedToGeneratePropertySource(Exception e) {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigDataLoader.java:162]: Why does this method have a return value?

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationBootstrapRegistrar.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    private static AppConfigurationReplicaClientFactory buildClientFactory(
+        AppConfigurationReplicaClientsBuilder clientBuilder,
+        AppConfigurationProperties properties) {
+        return new AppConfigurationReplicaClientFactory(clientBuilder, properties.getStores());
+    }
+
+    private static AppConfigurationReplicaClientsBuilder replicaClientBuilder(ConfigDataLocationResolverContext context,
+        Binder binder,
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationBootstrapRegistrar.java:52]: Seems like the binder is not used.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationBootstrapRegistrar.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        context.getBootstrapContext().registerIfAbsent(AppConfigurationKeyVaultClientFactory.class,
+            InstanceSupplier.from(() -> keyVaultClientFactory));
+        context.getBootstrapContext().registerIfAbsent(AppConfigurationReplicaClientFactory.class,
+            InstanceSupplier.from(() -> replicaClientFactory));
+    }
+
+    private static AppConfigurationKeyVaultClientFactory appConfigurationKeyVaultClientFactory(
+        AppConfigurationProviderProperties appProperties,
+        ConfigDataLocationResolverContext context, Binder binder)
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationBootstrapRegistrar.java:35]: Seems like the binder is not used.

**Pull request closed without merging**